### ISSUE TITLE
Castlevania Dominus Collection - 0100FA501AF90000

### ISSUE THREAD

**Issue by RupeeClock** [Ryujinx/Ryujinx-Games-List#5047]
### Game Name

Castlevania Dominus Collection

### Game Update Version

1.0.0

### Emulator Build Version

1.1.1380

### Graphics Backend

Vulkan

### Hardware Specs

- ##### CPU: *Intel Core i7-12700k*
- ##### GPU: *NVIDIA RTX 3070*
- ##### GPU Driver version: *560.81*
- ##### RAM: *32GB DDR5 (2x16)*

### Game State

Castlevania Dominus Collection is a collection of three Castlevania games originally released for Nintendo DS, as well as the arcade title Haunted Castle, and a brand new remake called Haunted Castle Revisited.
All titles (except the remake) make use of emulation and appear to play correctly at full speed (60 FPS) with no obvious visual issues.
Touch screen inputs function in both Docked and Handheld modes. The DS games feature enhanced resolution over the original releases, with 3D effects and sprite rotations and scaling having better detail.
In the game selection menu, opening videos encoded in MPEG-4 version 2 format will play in the background without issue.
The provided rewind feature works as intended.

Various screen layouts are provided, some of which provide integer or non-integer scaling with nearest-neighbour resampling, where scaling can differ between docked and handheld modes.
Increasing the internal resolution can help improve the look of scaling in some instances, but can introduce visible seams in some sprites elements.

### Screenshots

![Screenshot (292)](https://github.com/user-attachments/assets/441ec8a3-9685-407d-a5de-335d1f3c89b2)
![Screenshot (293)](https://github.com/user-attachments/assets/c773ce90-12cc-4a6a-a0fd-ad6ab46a3119)
![Screenshot (294)](https://github.com/user-attachments/assets/04ca9ec6-ba4d-4bdd-83f7-0fd4f1e74bfb)
![Screenshot (295)](https://github.com/user-attachments/assets/e4440a81-3763-4eb7-a28a-38b34bc6428c)
![Screenshot (296)](https://github.com/user-attachments/assets/a707b201-ff97-446e-a1a2-dc40e9adbe52)
![Screenshot (299)](https://github.com/user-attachments/assets/6392498a-97f3-492e-9f52-de87da9e886c)
![Screenshot (304)](https://github.com/user-attachments/assets/fd2c4148-7c7f-4ca0-bf16-61fabc3c49e6)
![Screenshot (305)](https://github.com/user-attachments/assets/683b7548-9067-4eb8-8eeb-63c2a5673ab5)
![Screenshot (306)](https://github.com/user-attachments/assets/51832050-bd72-4bc4-a24a-a49f667e52df)
![Screenshot (307)](https://github.com/user-attachments/assets/507201da-6ffe-41c8-bf8d-98201ca4228e)

### Log File

[Ryujinx_1.1.1380_2024-08-28_22-08-39.log](https://github.com/user-attachments/files/16789092/Ryujinx_1.1.1380_2024-08-28_22-08-39.log)

**Comment by Veltiel** [comment]
Game Name
Castlevania Dominus Collection

Game Update Version
1.0.0

Emulator Build Version
1.1.1380

Graphics Backend
Vulkan

Hardware Specs
CPU: Intel Core i5-9400f
GPU: NVIDIA RTX 3060 12 GB
GPU Driver version: 560.81
RAM: 16GB DDR4 (2x8)

Sadly, in my environment this anthology seems to go slow. I only have from 20 to 30 FPS, and is nowhere near as fast as playing either Castlevania game on a DS Emulator. At least while playing Dawn of Sorrow or Order of Eclessia on an DS Emulator, it goes x5 times faster than emulating this anthology on RyujinX.  And It seems I can't skip the intro in neither Castlevania game. I have to wait until the gameplay starts and it goes slow, the same with the dialogue. Am I the only one experiencing this?

**Comment by antoniodesousa** [comment]
Yes, it runs slow for me as well. And I see some graphical bugs in some places. Like right at the beginning in Down of Sorrow some parts of the buildings are pink. Yuzu runs much better without any graphical issues but still not full speed.

Ryujinx build 1.1.1385
CPU: Ryzen 9 6900HX
GPU: Radeon RX 6600M
RAM: 64 DDR5
OS: Windows 11 Pro 23H2

**Comment by ZongMan** [comment]
Getting the same problems in Update 1.21.0 (v1376256) as well. I'm using the always latest updated Ryujinx

**Comment by Veltiel** [comment]
> Yes, it runs slow for me as well. And I see some graphical bugs in some places. Like right at the beginning in Down of Sorrow some parts of the buildings are pink. Yuzu runs much better without any graphical issues but still not full speed.
> 
> Ryujinx build 1.1.1385 CPU: Ryzen 9 6900HX GPU: Radeon RX 6600M RAM: 64 DDR5 OS: Windows 11 Pro 23H2

Wait, Yuzu? I have the latest prod keys and firmware update on it, but as the emulator is not longer updated, the game just won't boot, even if the emulator recognizes the altest prod keys. How did you even manage to run it?

**Comment by antoniodesousa** [comment]
> Sadly, in my environment this anthology seems to go slow. I only have from 20 to 30 FPS, and is nowhere near as fast as playing either Castlevania game on a DS Emulator. At least while playing Dawn of Sorrow or Order of Eclessia on an DS Emulator, it goes x5 times faster than emulating this anthology on RyujinX. And It seems I can't skip the intro in neither Castlevania game. I have to wait until the gameplay starts and it goes slow, the same with the dialogue. Am I the only one experiencing this?

Yes, it runs slow for me as well. And I see some graphical bugs in some places. Like right at the beginning in Down of Sorrow some parts of the buildings are pink.

Ryujinx build 1.1.1385
CPU: Ryzen 9 6900HX
GPU: Radeon RX 6600M
RAM: 64 DDR5
OS: Windows 11 Pro 23H2

**Comment by GamerzHell9137** [comment]
Should be noted that on Windows the game is playing slow on most CPUs.
On a 7800X3D you're getting max 74 FPS with Vsync off and on a 5500 you're getting around 45 FPS which is under the games 60 FPS speed.

**Comment by RupeeClock** [comment]
Following version 1.1.1388, FIFO pipeline measurements when playing the DS games in Dominus Collection has lowered drastically on my machine.
The starting area of Portrait of Ruin, which previously had FIFO around 60%, is now down to around 3%.
![image](https://github.com/user-attachments/assets/f4a15940-2a1a-4202-8b70-cc144afa0d81)
